app/tests/integration/test_generic_experiment_e2e.py

```diff
@@ -0,0 +1,84 @@
+from urllib.parse import urlparse
+
+import pytest
+import requests
+
+from models.validation_dataclass import APIDataClass
+from models.base_dataclass import (
+    BasePreferencesDataClass,
+    BaseBranchDataClass,
+    BaseDataClass,
+)
+
+
+@pytest.fixture(name="experiment_type", scope="module")
+def fixture_experiment_type():
+    return "single-pref-experiment"
+
+
+@pytest.fixture(name="experiment_name", scope="module")
+def fixture_experiment_name():
+    return "Generic Experiment"
+
+
+@pytest.fixture(name="default_data", scope="module")
+def fixture_default_data():
+    branches = []
+    preferences = [
+        BasePreferencesDataClass(
+            preference_branch_name="e2e-testing",
+            preference_branch_type="default",
+            preference_type="boolean",
+            preference_value="true",
+        ),
+        BasePreferencesDataClass(
+            preference_branch_name="e2e-testing",
+            preference_branch_type="default",
+            preference_type="boolean",
+            preference_value="false",
+        ),
+    ]
+
+    for count, item in enumerate(preferences):
+        branches.append(
+            BaseBranchDataClass(
+                branch_name=f"e2e-generic-branch-{count}", preferences=item
+            )
+        )
+
+    return BaseDataClass(
+        type_name="Generic Experiment",
+        action_name=None,
+        experiment_type="channel",
+        channels="Nightly",
+        min_version=99,
+        max_version=100,
+        user_facing_name="e2e testing generic experiment",
+        user_facing_description="e2e testing generic description",
+        branches=sorted(branches, key=lambda x: x.branch_name),
+    )
+
+
+@pytest.mark.use_variables
+@pytest.mark.nondestructive
+def test_generic_experiment_e2e(
+    base_url,
+    selenium,
+    experiment_type,
+    fill_timeline_page,
+    fill_design_page_generic_experiment,
+    fill_analysis_page,
+    fill_risks_page,
+    signoff_and_ship,
+    variables,
+    default_data,
+):
+    url = urlparse(selenium.current_url)
+    experiment_url = f"{url.scheme}://{url.netloc}/api/v1{url.path}recipe/"
+    experiment_json = requests.get(f"{experiment_url}", verify=False).json()
+    api_json = APIDataClass(**experiment_json)
+    assert default_data.action_name == api_json.action_name
+    assert default_data.experiment_type == api_json.filter_object[1].type
+    assert default_data.channels.lower() in api_json.filter_object[1].channels
+    assert default_data.min_version == api_json.filter_object[2].versions[0]
+    assert default_data.max_version == api_json.filter_object[2].versions[-1]
```

Comment:
Were you saying https://pydantic-docs.helpmanual.io/ would help avoid a big list of assertions? Or does that not work here because we're calling `.lower()` and `versions[-1]`, etc?

Reply:
Yeah so with this pydantic doesn't work like in my demo because the api is in a weird structure compared to the base dataclass. And it doesn't make sense trying to match it, so we cant just do ```assert obj == obj```. There are a bunch of little issues on why we cant do it how I showed it.

Reply:
We will probably iterate on it though, so if you have an idea please open an issue @pdehaan

Reply:
Yeah @jrbenny35 and I spent a bunch of time trying to figure out how to get them to match but there wasn't an obvious good way so I think this is still totally fine.